### DISCUSSION TITLE
Make global filter use column custom filters

### DISCUSSION THREAD

**Discussion by gunnartorfis** [TanStack/table#2649] [General]
A table uses a custom filtering just for one column.

How do I make the global filter use the custom filter for that column but the default built-in one (text) for the remaining columns?

**Comment by lkostrowski** [comment]
@tannerlinsley can I bump this one?

I have similar problem - my column data is JSON, I can make custom filter algorithm with `column.filter`, but it works only for colum-only filter, not global one

**Comment by bsubba** [comment]
May be I am too late here.But this was how I created custom global filter:
``` 
const filterTypes = React.useMemo(
    () => ({
      customFilterName: (rows, columnIds, filterValue) => {
       // custom logic
        });
      }
    }),
    []
  );

  const instance = useTable(
    {
      columns,
      data,
      ...
     ... ,
      globalFilter: customFilterName,
      filterTypes
    },
    useGlobalFilter
  );

**Reply by lkostrowski** [reply]
filter works when accessor is stringified json for me

**Comment by EricBrown3** [comment]
On first reading the docs, I thought this *was* the behavior of filterFn on the column def. But no, filterFn is only for the column's individual filter. 

It seems what's needed is a `globalFilterFn` on the column def, which allows a per-column global filter fn which overrides the table one.
